Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberView.swift

```diff
@@ -70,8 +105,11 @@ extension CardNumberView: TextFieldViewDelegate {
 
     let text = oldValue.replacingCharacters(in: textRange, with: string)
 
-    if let newTextFieldValue = viewModel.textFieldUpdate(from: text) {
+    if let (newTextFieldValue, schemeIcon) = viewModel.eagerValidate(cardNumber: text) {
```

Comment:
Should we add some check for text count before we send to eager validate?

Thinking to maybe have
```swift
let eagerValidateLenghtRange = 3...6
if eagerValidateLenghtRange.contains(text.count),
  (newTextFieldValue, schemeIcon) = viewModel.eagerValidate(cardNumber: text) {
```

Not sure if it makes sense, let me know what you think. My main reasoning was that once _validated eagerly_, it might not make sense to keep validating. So if I enter 4 characters and you have the eager validation, whilst I keep adding characters we could save the UI updates 🤔

Reply:
I agree that we should be minimising UI updates, but there are a few edge cases here that block us from only checking in the first few digits or only checking until we have a scheme:

## Scenario A

1. Enter full Visa card number
1. Select all and paste in Amex card number

This would result in the wrong scheme icon shown, wrong formatting, and the wrong security code validation length

## Scenario B

1. Enter Mada card number

As Mada card numbers are a subset of Visa and Mastercard card numbers, the card will first register as a Visa/Mastercard and then when we have enough characters it will change to Mada. eg. 5297412542005689

Reply:
Makes sense, thanks for explaining it 👍